src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import analysis.fourier.fourier_transform
+import measure_theory.integral.exp_decay
+import analysis.special_functions.gaussian
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), 𝓕 f n`, where `𝓕 f` is the
+Fourier transform of `f`, under the following hypotheses:
+* `f` is a continuous function `ℝ → ℂ`.
+* The sum `∑ (n : ℤ), 𝓕 f n` is convergent.
+* For all compacts `K ⊂ ℝ`, the sum `∑ (n : ℤ), sup { ‖f(x + n)‖ | x ∈ K }` is convergent.
+
+The last condition is automatic if we assume `f` has sufficiently rapid decay (e.g. if it is a
+Schwartz function); this will be added in future.
+-/
+
+noncomputable theory
+
+open function (hiding comp_apply) complex real set topological_space filter measure_theory
+
+open_locale real big_operators filter fourier_transform
+
+local attribute [instance] real.fact_zero_lt_one
+
+namespace continuous_map
+
+variables
+  {α : Type*} [topological_space α]
+  {E : Type*} [normed_add_comm_group E]
+
+/-- The continuous map `λ x, x * m`. -/
+@[to_additive "The continuous map `λ x, x + m`", simps]
+def mul_right [has_mul α] [has_continuous_mul α] (m : α) : C(α, α) := mk _ (continuous_mul_right m)
+
+/-- Summing the translates of `f` by `ℤ • p` gives a map which is periodic with period `p`.
+(This is true without any convergence conditions, since if the sum doesn't converge it is taken to
+be the zero map, which is periodic.) -/
+lemma periodic_tsum_comp_add_zsmul
+  [locally_compact_space α] [add_comm_group α] [topological_add_group α]
+  (f : C(α, E)) (p : α) :
+  periodic ⇑(∑' (n:ℤ), f.comp (add_right $ n • p)) p :=
```

Comment:
I assume this should be moved to an earlier file?

Reply:
I'm not so sure. This PR has already spun off a whole load of little sub-PRs with modifications to other files; but for this particular lemma, I have my doubts whether it is actually interesting outside this very specific context.

Reply:
You would be surprised how often I thought that, moved the lemma anyway, and then saw someone use it in a context I had no idea about!

In this case, I expect ergodic theory to eventually want some version of this lemma.

Re the spun off PRs. I'm sorry, this is what pushing the boundaries of mathlib feels like: you write the code in 2 days, and then it takes a year to merge all the prerequisites in the appropriate setting and surrounding API (eg Krein-Milman).

Reply:
> Re the spun off PRs. I'm sorry, this is what pushing the boundaries of mathlib feels like: you write the code in 2 days, and then it takes a year to merge all the prerequisites in the appropriate setting and surrounding API (eg Krein-Milman).

No worries, I know the score by now. 

Not sure where the best place for this lemma would be, though -- maybe in `topology.continuous_function.algebra`?